core/etcd/config/stack_config.go

```diff
@@ -0,0 +1,83 @@
+package config
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+
+	controlplaneconfig "github.com/kubernetes-incubator/kube-aws/core/controlplane/config"
+	"github.com/kubernetes-incubator/kube-aws/filereader/jsontemplate"
+	"github.com/kubernetes-incubator/kube-aws/logger"
+	"github.com/kubernetes-incubator/kube-aws/model"
+)
+
+// StackConfig contains configuration parameters available when rendering CFN stack template from golang text templates
+type StackConfig struct {
+	*controlplaneconfig.Config
+	StackName string
+	controlplaneconfig.StackTemplateOptions
+	UserDataEtcd      model.UserData
+	ExtraCfnResources map[string]interface{}
+	model.EtcdExistingState
```

Comment:
This and func NewStackConfig are the only changes from the controlplane version.